src/include/mpir_handlemem.h

```diff
@@ -57,15 +57,25 @@ int MPIR_check_handles_on_finalize(void *objmem_ptr);
 
 /* This routine is called by finalize when MPI exits.  This routine may be called
  * before initialization and after finalization of MPI. */
-static inline int MPIR_Handle_free(void **indirect, int indirect_size)
+static inline int MPIR_Handle_free(void **indirect, int indirect_size, bool no_mpl)
 {
+    /* Some handles (e.g., MPI_Info) are allocated by malloc(), not MPL_malloc(), so we
+     * need to deallocate them by free(). */
     int i;
 
     /* Remove any allocated storage */
     for (i = 0; i < indirect_size; i++) {
-        MPL_free((indirect)[i]);
+        if (no_mpl) {
+            free((indirect)[i]);
+        } else {
+            MPL_free((indirect)[i]);
+        }
+    }
+    if (no_mpl) {
+        free(indirect);
+    } else {
+        MPL_free(indirect);
     }
-    MPL_free(indirect);
```

Comment:
`MPIR_Handle_free()` is called whenever all the indirect `MPI_Info` handles are freed, so this function must be changed.

Reply:
Oh I see, I missed the new code you are adding. 

`MPIR_Handle_free` is very short, so I definitely would like to see two separate versions of this function rather than making this function more complicated. We cannot use `MPIR_Handle_free_impl` since `_impl` suffix are reserved as the glue layer between `MPI_` and `MPIR_`. How about `MPIR_Info_mem_free` or `MPIR_Info_handle_free`? I don't think the `Handle` part of the name makes sense since it confuses with single handle object.

Reply:
Thanks, okay, then I will choose `MPIR_Info_mem_free()` (and rename `MPIR_Handle_free` to `MPIR_Mem_free`).

Reply:
> and rename MPIR_Handle_free to MPIR_Mem_free

How about `MPIR_handle_mem_free`?

Reply:
Sounds good. Then I will use `MPIR_Info_handle_mem_free()`.

Reply:
Alright :)

Reply:
Now `MPIR_Handle_free()` takes `MPIR_Object_alloc_t` as an argument, so `MPIR_Handle_free()` can handle everything (including checking if it is a MPIR_INFO or not); I do not think we need another function, `MPIR_Info_handle_mem_free()` considering the current implementation of `MPIR_Handle_free()`.

Because `MPIR_Info_handle_mem_free()` is not newly added, I did not rename `MPIR_Handle_free()` to `MPIR_Handle_mem_free()`